constructor/fcp.py

```diff
@@ -336,14 +337,24 @@ def _main(name, version, download_dir, platform, channel_urls=(), channels_remap
                 new_file_name = "%s%s" % (dist[:-8], transmute_file_type)
                 new_dists.append(new_file_name)
                 new_file_name = os.path.join(download_dir, new_file_name)
-                if os.path.exists(new_file_name):
-                    continue
-                print("transmuting %s" % dist)
-                conda_package_handling.api.transmute(os.path.join(download_dir, dist),
-                    transmute_file_type, out_folder=download_dir)
+                if not os.path.exists(new_file_name):
+                    print("transmuting %s" % dist)
+                    failed_files = conda_package_handling.api.transmute(
+                        os.path.join(download_dir, dist),
+                        transmute_file_type,
+                        out_folder=download_dir,
+                    )
+                    if failed_files:
+                        raise RuntimeError("Transmution failed:", failed_files)
+                url, md5 = _urls[dist]
+                assert url.endswith(".tar.bz2")
```

Comment:
This is code smell since assert statements are easily disabled in production environments and shouldn't be used for validation. If you want this to fail loudly, please add a condition and fail with a RuntimeError like above.

Reply:
I'm pretty sure this is impossible and even if it fails I'm fairly sure the code would still work. If you'd rather not have `assert`s I propose drop it entirely:
```suggestion
```

Reply:
Yeah, unfortunately there are a few environments in which people specify [PYTHONOPTIMIZE](https://docs.python.org/using/cmdline.html#envvar-PYTHONOPTIMIZE) or `-O` in their build environment, which disables assert statements. It's quite the bummer. Let's remove the line then.